operators/src/machine_learning/onnx.rs

```diff
@@ -258,18 +295,41 @@ where
                 }
 
                 let pixels = pixels.into_iter().flatten().collect::<Vec<TIn>>();
-                let rows = width * height;
-                let cols = num_bands;
 
-                let samples = Array2::from_shape_vec((rows, cols), pixels).expect(
-                    "Array2 should be valid because it is created from a Vec with the correct size",
-                );
-
-                let input_name = &session.inputs[0].name;
-
-                let outputs = session
-                    .run(ort::inputs![input_name => samples].context(Ort)?)
-                    .context(Ort)?;
+                let outputs = if self.model_metadata.input_is_single_pixel() {
```

Comment:
could/should we have an enum of shapes and match here instead of if else if else?

Reply:
its only two cases and a default that results in an error. I don't know how an enum will make it less complex.